### ISSUE TITLE
Query about ClientCredentials

### ISSUE THREAD

**Issue by chrismatchett** [bshaffer/oauth2-server-php#741]
Hi,

I've been working through the demo, much appreciated thanks.

I stumbled across the fact UserCredentials and PDO storage defaults to looking for passwords in SHA-1.

Is there a config option which allows me to have the same SHA-1 behaviour for passwords using UserCredentials.

Thanks again,

Chris

**Comment by CameronGo** [comment]
Hey Chris, I'm not sure if this helps you.  I'm not using the Users table b/c we are doing authentication against an external system, but here is what I implemented to store ClientCredentials using the password_hash and password_verify functions.

#720

**Comment by chrismatchett** [comment]
Thanks CameronGo,

I will use your solution as a starting point.

I know I'm missing something but it's still curious to me why Google/Facebook don't hash these passwords...I'm sure the penny will drop at some point.

Thanks again,

Chris

**Comment by CameronGo** [comment]
I agree.  It is defined as a secret, but not treated as such.  I've opted to treat mine as a true password (as you can see) including creating a page to allow them to reset their secret.  So our workflow is to create an initial secret and then direct them to a page to change it before they use it in production.
